cas-client-core/src/main/java/org/jasig/cas/client/validation/AbstractTicketValidationFilter.java

```diff
@@ -186,6 +196,26 @@ protected void onFailedValidation(final HttpServletRequest request, final HttpSe
         // nothing to do here.
     }
 
+    private void regenerateSession(HttpServletRequest request) {
+        HttpSession oldSession = request.getSession();
+        Enumeration<String> attrNames = oldSession.getAttributeNames();
+        Properties props = new Properties();
+
+        if (attrNames != null) {
```

Comment:
You don't change the session identifier if there is no attributes, do you?

I guess it should always be done (changing the session identifier). I think it's the way it works in Spring Security.